tiles/src/main/java/com/protomaps/basemap/layers/Roads.java

```diff
@@ -22,7 +22,24 @@ public String name() {
   public void processFeature(SourceFeature sourceFeature, FeatureCollector features) {
     if (sourceFeature.canBeLine() && sourceFeature.hasTag("highway") &&
       !(sourceFeature.hasTag("highway", "proposed", "abandoned", "razed", "demolished", "removed", "construction"))) {
+
       String highway = sourceFeature.getString("highway");
+      String shield_text = sourceFeature.getString("ref");
+      String network_val = sourceFeature.getString("network");
+      shield_text = (shield_text == null ? null : shield_text.split(";")[0].replaceAll("\\s", ""));
+      if( shield_text != null ) {
+        if( shield_text.contains("US ") ) {
+          shield_text = shield_text.replaceAll("US ", "");
+          network_val = "US:US";
+        } else if( shield_text.contains("I ") ) {
+          shield_text = shield_text.replaceAll("I ", "");
+          network_val = "US:I";
+        } else {
+          network_val = "other";
+        }
+      }
+      Integer shield_text_length = (shield_text == null ? null : shield_text.length());
```

Comment:
Once we've done all our transforms on the string, we calculate the shield_text_length for later export as `ref_length`... This makes styling in MapLibre much easier.